robot_calibration/src/finders/plane_finder.cpp

```diff
@@ -319,6 +322,57 @@ sensor_msgs::PointCloud2 PlaneFinder::extractPlane(sensor_msgs::PointCloud2& clo
   return plane_cloud;
 }
 
+int sampleCloud(const sensor_msgs::PointCloud2& points,
+                double sample_distance, size_t max_points,
+                std::vector<geometry_msgs::PointStamped>& sampled_points)
+{
+  // Square distance for efficiency
+  double max_dist_sq = sample_distance * sample_distance;
+
+  // Iterate through cloud
+  sensor_msgs::PointCloud2ConstIterator<float> points_iter(points, "x");
+  for (size_t i = 0; i < points.width; ++i)
+  {
+    // Get (untransformed) 3d point
+    geometry_msgs::PointStamped rgbd;
+    rgbd.point.x = (points_iter + i)[X];
+    rgbd.point.y = (points_iter + i)[Y];
+    rgbd.point.z = (points_iter + i)[Z];
+
+    // Is this far enough from our current points?
+    bool include_point = true;
+    for (auto point : sampled_points)
```

Comment:
Just a thought: it may be worth adding an optimization that keeps track of the sampled_points' xyz lower and upper bounds... this way if a new point is outside the bounds with a dist > max_dist_sq, we can break knowing that no sampled point is close-by.
However, it doesn't make sense to add this if our cloud isn't that big...